### PR TITLE
fix: override_if_not_empty should not skip falsy primitives (0, False, 0.0)

### PR DESCRIPTION
**Pull request by binggao1230**
## Problem

`TypeConflictStrategies.strategy_override_if_not_empty` uses a bare truthiness check:

```python
return nxt if nxt else base
```

This treats *any* falsy value as "empty", including `0`, `False`, and `0.0`.
These are valid, intentional values — they are **not** empty, and they are **not**
null — so they should override `base`.

```python
from deepmerge import Merger
m = Merger([], "override", "override_if_not_empty")

m.merge("base", 0)      # returns "base"  ← wrong, should be 0
m.merge("base", False)  # returns "base"  ← wrong, should be False
m.merge("base", 0.0)    # returns "base"  ← wrong, should be 0.0
```

## Fix

Replace the truthiness check with an explicit null/empty test:

1. `None` → keep `base` (null)
2. `len(nxt) == 0` (dict, list, set, str, …) → keep `base` (empty container)
3. Anything else (including `0`, `False`, `0.0`) → return `nxt`

The empty check is `isinstance(nxt, Sized) and len(nxt) == 0`, so non-sized
types (ints, floats, bools) fall through to the override path.

The existing tests pass unchanged; a parameterized test covers the falsy
primitives and the empty/null cases.

### CODE REVIEW

File: deepmerge/tests/strategy/test_type_conflict.py [hunk 1]
```diff
@@ -20,3 +20,15 @@ def test_merge_if_not_empty():
...
+def test_merge_if_not_empty_falsy_primitives():
+    """Falsy primitives (0, False) are valid non-empty values and should override base."""
+    strategy = TypeConflictStrategies.strategy_override_if_not_empty({}, [], "base", 0)
+    assert strategy == 0, "integer zero is not empty; it should override base"
+
+    strategy = TypeConflictStrategies.strategy_override_if_not_empty({}, [], "base", False)
+    assert strategy is False, "False is not empty; it should override base"
+
+    strategy = TypeConflictStrategies.strategy_override_if_not_empty({}, [], "base", 0.0)
+    assert strategy == 0.0, "0.0 is not empty; it should override base"
```
Comment (toumorokoshi) [inline deepmerge/tests/strategy/test_type_conflict.py]: Can you convert this to a parameterized test?

https://docs.pytest.org/en/stable/example/parametrize.html

that way it avoids a quit exit without testing the other edge cases.

**Review by toumorokoshi — CHANGES_REQUESTED** [review]
Thanks! this looks right, but a quick comment.

Also although I agree this is a "fix", I think we'll do a major version bump - it's a breaking change to a merge strategy.

File: deepmerge/tests/strategy/test_type_conflict.py (hunk 1 above)
Comment (binggao1230) [reply deepmerge/tests/strategy/test_type_conflict.py]: Done in `e16063a` — 13 parameterized cases with ids.

**Comment by binggao1230** [comment]
Parameterized in `e16063a` — 13 cases with ids, covering the falsy primitives plus the empty/null side so both directions are in one table. Your point about the early exit is exactly what shows up: against master the run reports all four broken cases instead of stopping at the first one.

```
FAILED test_merge_if_not_empty_falsy[zero int]
FAILED test_merge_if_not_empty_falsy[false]
FAILED test_merge_if_not_empty_falsy[zero float]
FAILED test_merge_if_not_empty_falsy[zero complex]
4 failed, 9 passed
```

It asserts the type as well as the value, since `0 == False` would otherwise let a wrong return slip through. Whole suite is 34 passed, `black --check` and `mypy` clean.

Major version bump is your call, that makes sense to me. Also corrected the PR description: it still claimed a `try/except TypeError`, which `07e8b4f` replaced with the `Sized` check.

**Comment by toumorokoshi** [comment]
great, thank you!
